scripts/windows/test/Run-E2ETest.ps1

```diff
@@ -0,0 +1,471 @@
+            -e `"$EventHubConnectionString`" ``
+            -r `"edgebuilds.azurecr.io`" ``
+            -u `"$ContainerRegistryUsername`" ``
+            -p `"$ContainerRegistryPassword`" --verify-data-from-module `"tempFilter`" ``
+            -t `"${ReleaseArtifactImageBuildNumber}-windows-$(GetImageArchitectureLabel)`" ``
+            -l `"$ModuleToModuleDeploymentWorkingFilePath`""
+    $testCommand = AppendInstallationOption($testCommand)
+    Invoke-Expression $testCommand
+    $testExitCode = $lastExitCode
+
+    PrintLogs $testStartAt $testExitCode
+    exit $testExitCode
+}
+
+Function RunTempSensorTest
+{
+    PrintHighlightedMessage "Run TempSensor test"
+    TestSetup
+
+    $testStartAt = Get-Date
+    $deviceId = "e2e-${ReleaseLabel}-Windows-tempSensor"
+    PrintHighlightedMessage "Run quickstart test with -d ""$deviceId"" started at $testStartAt."
+
+    $testCommand = "&$IoTEdgeQuickstartExeTestPath ``
+        -d `"$deviceId`" ``
+        -c `"$IoTHubConnectionString`" ``
+        -e `"$EventHubConnectionString`" ``
+        -r `"edgebuilds.azurecr.io`" ``
+        -u `"$ContainerRegistryUsername`" ``
+        -p `"$ContainerRegistryPassword`" --optimize_for_performance true ``
+        -t `"${ReleaseArtifactImageBuildNumber}-windows-$(GetImageArchitectureLabel)`""
+    $testCommand = AppendInstallationOption($testCommand)
+    Invoke-Expression $testCommand
+    $testExitCode = $lastExitCode
+
+    PrintLogs $testStartAt $testExitCode
+    exit $testExitCode
+}
+
+Function RunTransparentGatewayTest
+{
+    PrintHighlightedMessage "Run Transparent Gateway test"
+    TestSetup
+
+    $testStartAt = Get-Date
+    $deviceId = "e2e-${ReleaseLabel}-Windows-QuickstartCerts"
+    PrintHighlightedMessage "Run quickstart test with -d ""$deviceId"" started at $testStartAt."
+
+    $testCommand = "&$IoTEdgeQuickstartExeTestPath ``
+        -d `"$deviceId`" ``
+        -c `"$IoTHubConnectionString`" ``
+        -e `"doesNotNeed`" ``
+        -n `"$env:computername`" ``
+        -r `"edgebuilds.azurecr.io`" ``
+        -u `"$ContainerRegistryUsername`" ``
+        -p `"$ContainerRegistryPassword`" --optimize_for_performance true ``
+        -t `"${ReleaseArtifactImageBuildNumber}-windows-$(GetImageArchitectureLabel)`" ``
+        --leave-running=Core ``
+        --no-verify ``
+        --device_ca_cert `"$DeviceCACertificatePath`" ``
+        --device_ca_pk `"$DeviceCAPrimaryKeyPath`" ``
+        --trusted_ca_certs `"$TrustedCACertificatePath`""
+    $testCommand = AppendInstallationOption($testCommand)
+    Invoke-Expression $testCommand
+
+    Write-Host "Run LeafDevice"
+    &$LeafDeviceExeTestPath `
+        -d "${deviceId}-leaf" `
+        -c "$IoTHubConnectionString" `
+        -e "$EventHubConnectionString" `
+        -ct "$TrustedCACertificatePath" `
+        -ed "$env:computername"
+    $testExitCode = $lastExitCode
+    
+    PrintLogs $testStartAt $testExitCode
+    exit $testExitCode
+}
```

Comment:
Thanks for this Script Philip. The only think I don't like about is it that a test case is a Function on this script. We are having a lot of duplication for small differences on theses tests. Can we think of a cleaner way? Suggestion: -Create a file (Can be a input test file), where each test case is a line on where the differences are just a comment line. Another idea is that a test case can be it's own File and this script would dynamically load these files (Like a Test Framework! :)). What do you think?

Reply:
Will refactor common code in another function.  Since the file is still small, so I will keep all tests in a single file for now.  And I will support to run all tests if no test name is provided.

Reply:
Support to run all tests option.  After review, existing common code in each tests are PrintHighlightedMessage, TestSetup and PrintLogs.  Refactoring these out doesn't get much help and I think existing implementation will provide feasibility and readability.  Therefore I will keep existing implementation.  Agreed to have a test framework; but this script is just to run e2e tests locally easier and improve code re-usability and maintainability (no inline code in VSTS pipeline) with minimum efforts.